src/abstract_charm.py

```diff
@@ -156,21 +186,60 @@ def wait_until_mysql_router_ready(self) -> None:
     #  Handlers
     # =======================
 
-    def reconcile_database_relations(self, event=None) -> None:
-        """Handle database requires/provides events."""
+    def _upgrade_relation_created(self, _) -> None:
+        if self._unit_lifecycle.authorized_leader:
+            # `self._upgrade.is_compatible` should return `True` during first charm
+            # installation/setup
```

Comment:
is this comment relevant? i do not see how `is_compatible` is related to the code in this method

Reply:
ah, that makes sense. may be just indicate that `is_compatible` will use the versions set in the databag, if you think it makes sense to add